File: src/main/java/net/flyingfishflash/loremlist/domain/LrmComponentType.java
package net.flyingfishflash.loremlist.domain;

public enum LrmComponentType {
  ListItem,
  Item,
  List;

  private LrmComponentType opposite;

  static {
    Item.opposite = List;
    List.opposite = Item;
    ListItem.opposite = ListItem;
  }

  public LrmComponentType invert() {
    return opposite;
  }
}

File: src/main/kotlin/net/flyingfishflash/loremlist/domain/LrmComponentType.kt
package net.flyingfishflash.loremlist.domain

enum class LrmComponentType {
  ListItem,
  Item,
  List,
  ;

  private lateinit var opposite: LrmComponentType

  fun invert(): LrmComponentType {
    return opposite
  }

  companion object {
    init {
      Item.opposite = List
      List.opposite = Item
      ListItem.opposite = ListItem
    }
  }
}
